src/cpufreq.c

```diff
@@ -82,9 +142,58 @@ static int cpufreq_read(void) {
     /* convert kHz to Hz */
     v.gauge *= 1000.0;
 
-    cpufreq_submit(i, v);
-  }
+    cpufreq_submit(i, "cpufreq", NULL, &v);
+
+    /* Read total transitions for cpu frequency */
+    if (report_total_trans) {
+      snprintf(filename, sizeof(filename),
+               "/sys/devices/system/cpu/cpu%d/cpufreq/stats/total_trans", i);
+      if (parse_value_file(filename, &v, DS_TYPE_COUNTER) != 0) {
+        WARNING("cpufreq plugin: Reading \"%s\" failed.", filename);
+        continue;
+      }
+      counter_t c = counter_diff(t_data[i].transition_prev, v.counter);
+      t_data[i].transition_prev = v.counter;
+      cpufreq_submit(i, "transitions", NULL, &(value_t){.counter = c});
+    }
 
+    /*
+     * Determine percentage time in each state for cpu during previous interval.
+     */
+    if (report_time_in_state) {
+      int j = 0;
+      char state[DATA_MAX_NAME_LEN], time[DATA_MAX_NAME_LEN];
+
+      snprintf(filename, sizeof(filename),
+               "/sys/devices/system/cpu/cpu%d/cpufreq/stats/time_in_state", i);
+      fh = fopen(filename, "r");
+      if (fh == NULL)
+        continue;
+      while (fgets(buffer, sizeof(buffer), fh) != NULL) {
+        if (!sscanf(buffer, "%s%lli", state, &t)) {
+          fclose(fh);
+          return 0;
+        }
+        snprintf(time, sizeof(time), "%lli", t);
+        if (parse_value(time, &v, DS_TYPE_DERIVE) != 0) {
+          WARNING("cpufreq plugin: Reading \"%s\" failed.", filename);
+          continue;
+        }
+        cdtime_t now = cdtime();
+        gauge_t g;
+        if (j < MAX_AVAIL_FREQS) {
+          if (value_to_rate(&g, v, DS_TYPE_DERIVE, now,
+                            &t_data[i].time_state[j]) != 0) {
+            continue;
+            j++;
+          }
+          cpufreq_submit(i, "percent", state, &(value_t){.gauge = g});
```

Comment:
Will this really be `percent`  - value in 0-100 range?

Reply:
I think so: the comment above says this is in 10ms increments, 10ms/1s is essentially a percentage.